### ISSUE TITLE
Write book builder

### ISSUE THREAD

**Issue by darribas** [pysal/notebooks#49]
Development for this is currently taking place at the [`nbs2.0`](https://github.com/darribas/notebooks/tree/nbs2.0) branch of @darribas fork. The roadmap is sketched under the [`nbs2.0`](https://github.com/pysal/notebooks/issues?q=is%3Aissue+is%3Aopen+label%3Anbs2.0) tag. Opening this as an umbrella issue for more general discussions about what we want to do. To mark as completed when the project goes live.

**Comment by slumnitz** [comment]
A first though on pulling readme files from single projects. `splot` has a picture in it's readme which is currently not displayed in the book. Is it generally possible to also pull figures or is there a way to embed the figure into the readme so the book also sees it?

**Comment by jGaboardi** [comment]
@darribas For adding `spaghetti` into the jupyterbook infrastructure, would I simply add it into the [`packages.yml`](https://github.com/darribas/notebooks/blob/ce02786d9128c03c4f34490f0108fd6ddb9f755c/lib/packages.yml#L1)?

pysal/notebooks#53

**Comment by darribas** [comment]
@jGaboardi Yes, that's right. Ideally, I want to get rid of that file and dynamically pull it from the PySAL meta package but, for now that it is. If you want to issue a PR, that'd be awesome. If no bandwidth, leave an issue at least and I'll get to it asap

@slumnitz Good catch! I'm not sure we can encode images on markdown cells. We could definitely pull in the images as now `build.py` first pulls the entire repo ([link](https://github.com/darribas/notebooks/blob/master/lib/build.py#L39)), but we'd need to find a clever way to do it automatedly, swaping the directory paths so they get recognised. Is there any standard practice on the package template for storing images (like a `fig` folder in the root folder)? If there was, it'd be fairly straightforward to pull them in as well.

**Comment by jGaboardi** [comment]
@darribas I have a commit ready to push up to [add spaghetti](https://github.com/pysal/notebooks/pull/53#issuecomment-451536815), but I am getting a permission denied error from `git`.

```
(py3_spgh_dev) wc-dhcp179d200:notebooks jgaboardi$ git push darribas darribas_master
Enter passphrase for key '/Users/jgaboardi/.ssh/id_rsa': xxxxxx
Counting objects: 4, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (4/4), done.
Writing objects: 100% (4/4), 1.01 KiB | 1.01 MiB/s, done.
Total 4 (delta 3), reused 0 (delta 0)
remote: Resolving deltas: 100% (3/3), completed with 3 local objects.
To github.com:darribas/notebooks.git
 ! [remote rejected] darribas_master -> darribas_master (permission denied)
error: failed to push some refs to 'git@github.com:darribas/notebooks.git'
```

**Comment by darribas** [comment]
That's right, I think you should fork my fork (`darribas/notebooks`) and issue a PR there, which I'll pull in. Alternatively, we can wait on this until #53 gets merged, and you can issue the PR directly to `pysal/notebooks`

**Comment by jGaboardi** [comment]
@darribas I was able to push up the commit to add spaghetti
